src/fabric.c

```diff
@@ -344,6 +344,8 @@ void fi_ini(void)
 {
 	char *param_val = NULL;
 
+	ofi_osd_init();
```

Comment:
I still don't like this call here.  Multiple threads can call fi_ini simultaneously, so the implementation of ofi_osd_init is very specific and requires internal synchronization to ensure that it only executes once.  That's exactly what the mutex below achieves.  So this call as it is located cannot be a generic osd initialization routine.

I understand that windows needs to initialize the mutex (critical section).  Can that be done in dll main - process attach instead?  Then add an osd specific init routine below where the other init calls are made.